src/DynamoCore/Library/DocumentationServices.cs

```diff
@@ -58,10 +58,18 @@ public static XmlReader GetForAssembly(string assemblyPath, IPathManager pathMan
                 assemblyLocation = cashedAssemblyLocation;
             }
 
-            var assemblyPath = Path.GetFullPath(assemblyLocation);
-
-            var baseDir = Path.GetDirectoryName(assemblyPath);
-            var xmlFileName = Path.GetFileNameWithoutExtension(assemblyPath) + ".xml";
+            string baseDir;
+
+            if (File.Exists(assemblyLocation))
+                // There are 2 cases: if assembly exists, then we will use assembly path and try 
+                // to find xml file there.
+                baseDir = Path.GetDirectoryName(Path.GetFullPath(assemblyLocation));
+            else
+                // If assembly does not exist, then we will use dynamo directory and try
+                // to find xml file there.
+                baseDir = pathManager.DynamoCoreDirectory;
+
+            var xmlFileName = Path.GetFileNameWithoutExtension(assemblyLocation) + ".xml";
```

Comment:
If we look at the documentation of `pathManager.ResolveLibraryPath` API, it does a pretty thorough job in locating the `assemblyLocation`, including inside `DynamoCoreDirectory`. If it cannot locate the file, the following becomes true:

```csharp
// Some nodes don't have assembly, e.g. operators, but they do have xml file.
if (String.IsNullOrEmpty(assemblyLocation))
{
}
```

I believe we can merge your fix with the above statement like so:

```csharp
if (pathManager != null)
{
    pathManager.ResolveLibraryPath(ref assemblyLocation);
}

var baseDir = string.Empty;
if (String.IsNullOrEmpty(assemblyLocation))
{
    // Some nodes do not have a corresponding assembly, but their documentation 
    // xml file resides alongside DynamoCore.dll. If the assembly could not be 
    // located, fall back onto using DynamoCoreDirectory.
    // 
    baseDir = pathManager.DynamoCoreDirectory;
}
else
{
    // Found the assembly location, search for documentation alongside it.
    baseDir = Path.GetDirectoryName(Path.GetFullPath(assemblyLocation));
}
```

This way we can remove the complexity of having to restore the `assemblyLocation`, causing the flow to be non-linear. Does that make sense?